src/protocol/attribution/apply_attribution_window.rs

```diff
@@ -0,0 +1,351 @@
+            Self::RandomBitsForBitDecomposition => "random_bits_for_bit_decomposition",
+            Self::TimeDeltaBitDecomposition => "time_delta_bit_decomposition",
+            Self::TimeDeltaLessThanCap => "time_delta_less_than_cap",
+            Self::CompareBitTimesTriggerValue => "compare_bit_times_trigger_value",
+        }
+    }
+}
+
+#[cfg(all(test, not(feature = "shuttle")))]
+mod tests {
+    use crate::attribution_window_test_input;
+    use crate::ff::{Field, Fp32BitPrime};
+    use crate::protocol::attribution::{
+        apply_attribution_window::apply_attribution_window,
+        input::{
+            ApplyAttributionWindowInputRow, MCApplyAttributionWindowInputRow,
+            MCApplyAttributionWindowOutputRow,
+        },
+    };
+    use crate::protocol::context::Context;
+    use crate::protocol::modulus_conversion::{convert_all_bits, convert_all_bits_local};
+    use crate::protocol::{BreakdownKey, MatchKey};
+    use crate::secret_sharing::replicated::semi_honest::AdditiveShare as Replicated;
+    use crate::secret_sharing::SharedValue;
+    use crate::test_fixture::input::GenericReportTestInput;
+    use crate::test_fixture::{Reconstruct, Runner, TestWorld};
+
+    #[tokio::test]
+    pub async fn attribution_window() {
+        const ATTRIBUTION_WINDOW: u32 = 600;
+        const EXPECTED: &[u128; 23] = &[
+            0, 0, 0, 10, 2, 1, 5, 1, 0, 0, 0, 10, 0, 3, 12, 0, 0, 6, 4, 0, 6, 1, 0,
+        ];
+
+        let input: Vec<GenericReportTestInput<Fp32BitPrime, MatchKey, BreakdownKey>> = attribution_window_test_input!(
+            [
+                { timestamp: 500, is_trigger_report: 0, helper_bit: 0, breakdown_key: 3, credit: 0 }, // delta: 0
+                { timestamp: 100, is_trigger_report: 0, helper_bit: 0, breakdown_key: 4, credit: 0 }, // delta: 0
+                { timestamp: 130, is_trigger_report: 0, helper_bit: 1, breakdown_key: 4, credit: 0 }, // delta: 0
+                { timestamp: 150, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 10 },// delta: 20
+                { timestamp: 250, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 2 }, // delta: 120
+                { timestamp: 310, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 1 }, // delta: 180
+                { timestamp: 420, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 5 }, // delta: 290
+                { timestamp: 540, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 1 }, // delta: 410
+                { timestamp: 890, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 3 }, // delta: 760
+                { timestamp: 920, is_trigger_report: 1, helper_bit: 1, breakdown_key: 0, credit: 5 }, // delta: 790
+                { timestamp: 110, is_trigger_report: 0, helper_bit: 0, breakdown_key: 1, credit: 0 }, // delta: 0
+                { timestamp: 310, is_trigger_report: 1, helper_bit: 0, breakdown_key: 0, credit: 10 },// delta: 0
```

Comment:
comment is wrong. delta is actually 200

Reply:
Gah... nm... helper_bit = 0, so it's just not the same user... Maybe instead of `delta: 0`, you could just say: `different match key`